Sources/SwiftKafka/KafkaClient.swift

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import Logging
+
+/// Base class for ``KafkaProducer`` and ``KafkaConsumer``, which is used to handle the connection to the Kafka ecosystem.
+class KafkaClient {
+    // Default size for Strings returned from C API
+    static let stringSize = 1024
+
+    let logger: Logger
```

Comment:
```suggestion
    /// A logger.
   private let logger: Logger
```

Reply:
This is still open

Reply:
My bad, have not noticed that it should be `private`